src/Compilers/Core/Portable/Syntax/InternalSyntaxNodeExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class InternalSyntaxNodeExtensions
+    {
+        /// <summary>
+        /// Gets the first node of type TNode that matches the predicate.
+        /// </summary>
+        internal static TNode? FirstAncestorOrSelf<TNode, TArg>(this SyntaxNode? node, Func<TNode, TArg, bool> predicate, TArg argument, bool ascendOutOfTrivia = true)
```

Comment:
>internal [](start = 8, length = 8)

Could external clients benefit from this method the same way as we benefit from it internally?

Reply:
This change was removed from the pull request.